packages/block-editor/src/hooks/block-style-variation.js

```diff
@@ -77,13 +78,25 @@ function OverrideStyles( { override } ) {
  * @return {JSX.Element|undefined} An array of new block variation overrides.
  */
 export function __unstableBlockStyleVariationOverridesWithConfig( { config } ) {
-	const { getBlockStyles, overrides } = useSelect(
-		( select ) => ( {
+	const { overrides, getBlockStyles } = useSelect( ( select ) => {
+		const { getBlockName, getStyleOverrides } = unlock(
+			select( blockEditorStore )
+		);
+		const _overrides = getStyleOverrides();
+		return {
+			overrides: _overrides?.length
+				? _overrides.map( ( [ id, override ] ) => [
```

Comment:
We can't return a new array in a selector unfortunately.

Reply:
Ah true. I knew it was too good to be true. I'll revert and leave this as an E2E update.